src/api/initialDataTypes.js

```diff
@@ -121,67 +121,67 @@ export type RawInitialDataRealmFilters = {|
  *
  * See notes on `RealmFilter` and `RealmLinkifier`.
  */
-export type InitialDataRealmFilters = {|
-  realm_filters: RealmFilter[],
-|};
+export type InitialDataRealmFilters = $ReadOnly<{|
+  realm_filters: $ReadOnlyArray<RealmFilter>,
+|}>;
 
-export type InitialDataRealmLinkifiers = {|
+export type InitialDataRealmLinkifiers = $ReadOnly<{|
   // Possibly absent: Not all servers can provide this. See
   // `InitialDataRealmFilters`.
-  realm_linkifiers?: RealmLinkifier[],
-|};
+  realm_linkifiers?: $ReadOnlyArray<RealmLinkifier>,
+|}>;
 
-export type RawInitialDataRealmUser = {|
+export type RawInitialDataRealmUser = $ReadOnly<{|
   avatar_source: 'G',
   avatar_url: string | null,
   avatar_url_medium: string,
   can_create_streams: boolean,
-  cross_realm_bots: Array<{| ...CrossRealmBot, avatar_url?: string | null |}>,
+  cross_realm_bots: $ReadOnlyArray<{| ...CrossRealmBot, avatar_url?: string | null |}>,
```

Comment:
Yet another layer of nesting: this `avatar_url` property isn't read-only :stuck_out_tongue: 

This is a good application for the `+` syntax -- where there's just one or two properties and the whole object type fits on a line, using `+` vs `$ReadOnly` can often save it from spilling into several lines.